src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -567,7 +581,7 @@ private Delegate MakeObjectGetter<T>(DataViewRow input, int iinfo, INamedOnnxVal
                     UpdateCacheIfNeeded(input.Position, srcNamedValueGetters, activeOutputColNames, outputCache);
                     var namedOnnxValue = outputCache.Outputs[_parent.Outputs[iinfo]];
                     var trueValue = namedOnnxValue.AsEnumerable<NamedOnnxValue>().Select(value => value.AsDictionary<string, float>());
-                    var caster = _parent.Model.ModelInfo.OutputsInfo[iinfo].Caster;
+                    var caster = _parent.Model.ModelInfo.OutputsInfo[mlnetOnnxMapping].Caster;
```

Comment:
The idea I suggested was that the new method would return the OutputInfo (not the new index), so that, for example in here and elsewhere were you used `mlnetOnnxMapping`, you could simply write `_parent.GetOutputInfo(iinfo).Caster;`. This way you could get rid of the `mlnextOnnxMapping` variable all together, in here and in the other method were you used it. I think it would be more readable in this way.

Reply:
Wouldn't that change the function signature?
Right now that function `MapDataViewColumnToOnnxOutputTensor` is called in only place. This function already has iinfo as a parameter. I think it is more consistent and readable to call the `MapDataViewColumnToOnnxOutputTensor` again here and retrieve the mapping instead of changing this function signature. 


---
In reply to: [434213871](https://github.com/dotnet/machinelearning/pull/5192#discussion_r434213871) [](ancestors = 434213871)

Reply:
In either case, it is a very minor change. @michaelgsharp I will leave it to you. This is a stylistic viewpoint that I won't insist on.

---
In reply to: [434215197](https://github.com/dotnet/machinelearning/pull/5192#discussion_r434215197) [](ancestors = 434215197,434213871)

Reply:
Precisely, the signature of this method here `MakeObjectGetter` was changed to pass the new parameter `mlnetOnnxMapping`, and now the `int iinfo` parameter which is also passed to this method is no longer used (so we might just as well remove it from here). Part of the idea of the refactoring I suggested was to not need to change the signature of this method so to still reuse the `iinfo` parameter. Also, wherever we use `mlnetOnnxMapping` we are also retrieving the OutputInfo. Using the new method to retrieve the OutputInfo instead of the index would be shorter.

But whatever @michaelgsharp  decides is good to me. 😉

Reply:
Precisely, the signature of this method here `MakeObjectGetter` was changed to pass the new parameter `mlnetOnnxMapping`, and now in this method we're using both the old infex `int iinfo` and the new one to access two different arrays, which might be a little bit confusing . Part of the idea of the refactoring I suggested was to not need to change the signature of this method so to still reuse the `iinfo` parameter. Also, wherever we use `mlnetOnnxMapping` we are also retrieving the OutputInfo; so using the new method to retrieve the OutputInfo instead of the index would be shorter.

But whatever @michaelgsharp  decides is good to me. 😉

Reply:
I decided to just call the method directly, so no changes to any function signatures are needed and it has been reverted back to the original signature.